### PR TITLE
Do not try to pull image if it exists

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java [hunk 1]
```diff
@@ -39,27 +40,11 @@ public DockerImageName apply(DockerImageName dockerImageName)
...
-                    if (progress != null) {
-                        System.err.println(padEnd(dockerImageName.asCanonicalNameString() + ":" + item.getId(), 50, ' ') + ' ' + progress);
-                    }
-                }
-            }).awaitCompletion();
-        }
-        catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            throw new RuntimeException("Could not pull image", e);
+        if (!imageExists(client, dockerImageName)) {
```
Comment (nineinchnick) [inline testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java:43]: `testcontainers` will always try to pull the image if it doesn't exist when resolving the image name, that is when `container.getDockerImageName()` gets called. I'm trying to address this in #16259.

Comment (nineinchnick) [reply testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java:43]: Maybe you could reuse the `DefaultPullPolicy` class from `testcontainers`, it's instance can be obtained from `org.testcontainers.images.PullPolicy.defaultPolicy()`

Comment (wendigo) [reply testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java:43]: This is solving some other issue - when the image is built locally and doesn't belong to any registry.

Comment (wendigo) [reply testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java:43]: It's orthogonal to your improvement in https://github.com/trinodb/trino/pull/16259.

Comment (nineinchnick) [reply testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java:43]: Thanks for clarifying! Please mark this as resolved, I can't.

**Review by kokosing — APPROVED** [review]

**Comment by colebow** [comment]
Does this need release notes? @wendigo cc @kokosing

**Comment by kokosing** [comment]
No. It is not UX affecting.

**Pull request merged**